source/options/OptionsState.hx

```diff
@@ -106,7 +107,16 @@ class OptionsState extends MusicBeatState
 
 		if (controls.BACK) {
 			FlxG.sound.play(Paths.sound('cancelMenu'));
-			MusicBeatState.switchState(new MainMenuState());
+			if (PlayState.instance != null && OptionsState.fromPlayState) //Check if player came from playstate.
```

Comment:
This line fixes the bug in the other PR where reentering/exiting optionsState from any state would load the last song played, regardless of whether it was finished or not.

Reply:
Pretty cool, adding that in

Reply:
It was already fixed…